iModelCore/ecobjects/PublicApi/ECObjects/ECSchemaConverter.h

```diff
@@ -175,10 +175,9 @@ struct ECSchemaConverter
     //! @param[in] schema   The schema to traverse
     //! @param[in] context  A context to use if a particular converter needs to locate a schema which is not currently a schema reference.
     //! @param[in] doValidate Flag saying whether to validate the schema or not.  This is used by the DgnV8Converter to disable validation until it has had a chance to fix the schemas
-    static bool Convert(ECSchemaR schema, ECSchemaReadContextP context = nullptr, bool doValidate = true)
+    static bool Convert(ECSchemaR schema, ECSchemaReadContextR context, bool doValidate = true)
         {
-        if (nullptr != context)
-            GetSingleton()->AddSchemaReadContext(*context);
+        GetSingleton()->AddSchemaReadContext(context);

```

Comment:
Mmh, this looks like it could be problematic. We have a global read context in a singleton which comes from "somewhere" and gets used "somewhere". I don't think read context was intended to be used like that. But I'm not sure if it can cause a problem when used like this.

Reply:
I do not even see the point of this PR if we going to keep a static schema cache. I thought the point was to remove the global static cache. This PR only changes how its initialized rather than removing it.

Reply:
The singleton pattern was used here to keep the original static version of the API available when we added a configurable instance version.  We can probably get rid of the static API and move entirely to an instance version so long as we update the connectors and keep their behavior consistent

Reply:
@AkshayN5 explain that to me. But my primary concern was` ~ECSchema()`dependency on global BeMutex. Having `SchemaReadContext` cached has that dependency that can cause a crash when the application exit when BeMutex is destroyed before the static schema cache. C++ does not ensure in which order in which global objects are destroyed and we saw the issue on MacOS/Linux where the test suit crashed on exit.

@AkshayN5 seems to have resolved my concern in his recent push by not caching read the context in a static member.

Reply:
@khanaffan Yes, I removed the read context from the static member in the recent commit. Schema context is now passed to the required functions as a parameter.